back/boxtribute_server/business_logic/box_transfer/agreement/mutations.py

```diff
@@ -19,16 +18,18 @@
 @mutation.field("createTransferAgreement")
 @convert_kwargs_to_snake_case
 def resolve_create_transfer_agreement(*_, creation_input):
-    # Enforce that the user can access at least one of the specified source bases
-    # (default: all bases of the user's organisation)
-    base_ids = creation_input.get(
-        "source_base_ids",
-        [
-            b.id
-            for b in Base.select().where(Base.organisation == g.user.organisation_id)
-        ],
-    )
+    # "Source" input arguments refer to the organisation/bases of the party that
+    # initiates the agreement.
+    try:
+        # Enforce that the user can access at least one of the bases of the initiating
+        # party (default: all bases of the user's organisation)
+        base_ids = creation_input.get(
+            "source_base_ids", g.user.authorized_base_ids("transfer_agreement:create")
+        )
+    except KeyError:
+        base_ids = []
     authorize(permission="transfer_agreement:create", base_ids=base_ids)
+
     return create_transfer_agreement(**creation_input, user=g.user)
```

Comment:
If I send a mutation request to the backend and do not specify `source_base_ids` and/or `target_base_ids` then the entry in the ´transfer_agreement_detail` table is NULL for either `source_base_id` and/or `target_base_id`. 

Should it not rather create entries in  ´transfer_agreement_detail` of all combinations of the bases a user has access to in `source_base_id`?

It just feels like there might be a small security hole. If a coordinator of Athens from BoxCare sends a `createTransferAgreement` request without `source_base_ids` being specified, it feels like it creates a transfer agreement for the whole organisation even though the coordinator can only speak for Athens.

If that is a problem, then possible solutions are:
- you pass somehow `g.user.authorized_base_ids` to create_transfer_agreement or
- we slowly start to make `source_base_ids` and/or `target_base_ids` required fields in the graphql query?? even though we then need to think how to change the UI again.

Reply:
Summary from discussion:
- make input for initiating base mandatory
- implications for accepting agreements with NULL partner bases has to be discussed separately

Reply:
1. is implemented
2. Partner bases NULL is equivalent to all bases of the partner org, hence only a user with access to all these bases can accept the agreement